### PR TITLE
[HIPIFY][Tensor][feature] Introduce cuTensor support - Part 6

### PR DESCRIPTION
**Pull request by aakanksha555**

### CODE REVIEW

File: src/CUDA2HIP_TENSOR_API_types.cpp [hunk 1]
```diff
@@ -185,6 +224,39 @@ const std::map<llvm::StringRef, cudaAPIversions> CUDA_TENSOR_TYPE_NAME_VER_MAP {
...
+  {"CUTENSOR_PLAN_PREFERENCE_CACHE_MODE",              {CUTENSOR_2000,  CUDA_0,        CUDA_0,      }},
+  {"CUTENSOR_PLAN_PREFERENCE_INCREMENTAL_COUNT",       {CUTENSOR_2000,  CUDA_0,        CUDA_0,      }},
+  {"CUTENSOR_PLAN_PREFERENCE_ALGO",                    {CUTENSOR_2000,  CUDA_0,        CUDA_0,      }},
+  {"CUTENSOR_PLAN_PREFERENCE_KERNEL_RANK",             {CUTENSOR_2000,  CUDA_0,        CUDA_0,      }},
+  {"CUTENSOR_PLAN_PREFERENCE_JIT",                     {CUTENSOR_2000,  CUDA_0,        CUDA_0,      }},
+  {"cutensorAutotuneMode_t",                           {CUTENSOR_1200,  CUDA_0,        CUDA_0,      }},
+  {"CUTENSOR_AUTOTUNE_MODE_NONE",                      {CUTENSOR_2000,  CUDA_0,        CUDA_0,      }},
+  {"CUTENSOR_AUTOTUNE_MODE_INCREMENTAL",               {CUTENSOR_2000,  CUDA_0,        CUDA_0,      }},
+  {"CUTENSOR_AUTOTUNE_NONE",                           {CUTENSOR_1200,  CUDA_0,        CUTENSOR_1700, }},
+  {"CUTENSOR_AUTOTUNE_INCREMENTAL",                    {CUTENSOR_1200,  CUDA_0,        CUTENSOR_1700, }},
```
Comment (emankov) [inline src/CUDA2HIP_TENSOR_API_types.cpp]: They are removed in 2.0.0.0, not in 1.7.0.0.

File: src/CUDA2HIP_TENSOR_API_types.cpp [hunk 2]
```diff
@@ -206,4 +278,7 @@ const std::map<llvm::StringRef, hipAPIversions> HIP_TENSOR_TYPE_NAME_VER_MAP {
   {"hiptensorAlgo_t",                                  {HIP_5070,       HIP_0,         HIP_0,       }},
   {"HIPTENSOR_ALGO_DEFAULT",                           {HIP_5070,       HIP_0,         HIP_0,       }},
   {"HIPTENSOR_ALGO_DEFAULT_PATIENT",                   {HIP_5070,       HIP_0,         HIP_0,       }},
+  {"HIPTENSOR_WORKSPACE_MIN",                          {HIP_5070,       HIP_0,         HIP_0,       }},
+  {"HIPTENSOR_WORKSPACE_RECOMMENDED",                  {HIP_5070,       HIP_0,         HIP_0,       }},
+  {"HIPTENSOR_WORKSPACE_MAX",                          {HIP_5070,       HIP_0,         HIP_0,       }},
```
Comment (emankov) [inline src/CUDA2HIP_TENSOR_API_types.cpp:284]: `hiptensorWorksizePreference_t` appeared in 5.7.0

**Review by emankov — CHANGES_REQUESTED** [review]

File: src/CUDA2HIP_TENSOR_API_types.cpp (hunk 2 above)
Comment (aakanksha555) [reply src/CUDA2HIP_TENSOR_API_types.cpp:284]: Done.

**Comment by aakanksha555** [comment]
Actually I'll need to regenerate the docs and hipify-perl as well. Let me do that too.

File: src/CUDA2HIP_TENSOR_API_types.cpp [hunk 3]
```diff
@@ -206,4 +278,7 @@ const std::map<llvm::StringRef, hipAPIversions> HIP_TENSOR_TYPE_NAME_VER_MAP {
   {"hiptensorAlgo_t",                                  {HIP_5070,       HIP_0,         HIP_0,       }},
   {"HIPTENSOR_ALGO_DEFAULT",                           {HIP_5070,       HIP_0,         HIP_0,       }},
   {"HIPTENSOR_ALGO_DEFAULT_PATIENT",                   {HIP_5070,       HIP_0,         HIP_0,       }},
+  {"HIPTENSOR_WORKSPACE_MIN",                          {HIP_5700,       HIP_0,         HIP_0,       }},
```
Comment (aakanksha555) [inline src/CUDA2HIP_TENSOR_API_types.cpp]: Please ignore this

**Review by emankov — APPROVED** [review]
LGTM

**Pull request merged**